components/match2/commons/match_group_util.lua

```diff
@@ -725,9 +754,12 @@ function MatchGroupUtil.matchIdFromKey(matchKey)
 	if round and matchInRound then
 		-- Bracket format
 		return 'R' .. string.format('%02d', round) .. '-M' .. string.format('%03d', matchInRound)
-	else
+	end
+
+	local matchIndex = matchKey:match('^M(%d+)$')
```

Comment:
If I'm not mistaken, this would require a change in all {{ShowSingleMatch}} template calls to MatchLists(including {{Single Match}}). From |matchid=1 to |matchid=M1.

Reply:
should make the `M` optional

Reply:
good catch